msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/ADAP3DFeatureDetectionMethod.java

```diff
@@ -93,11 +102,14 @@
 
     List<PeakDetection.GoodPeakInfo> newGoodPeakList =
         objPeakDetection.execute(objParameters, roundedFWHM);
-    List<SimpleFeature> newFeatureList = new ArrayList<SimpleFeature>();
-    newFeatureList.addAll(featureList);
-    newFeatureList.addAll(getFeature(newGoodPeakList));
-
-    return newFeatureList;
+    finalFeatureList = new ArrayList<SimpleFeature>();
```

Comment:
My question is, why don't you just keep adding peaks to ```featureList``` featureList? I think there is no need to create a new ArrayList.